catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java

```diff
@@ -467,7 +467,7 @@ private InputStream getInputStreamFromReference(Path externalReferencePath) thro
     Path newPath = null;
     if (reference.getScheme() == null) {
       newPath = Paths.get(reference.toASCIIString());
-    } else if (reference.getScheme().equalsIgnoreCase("file")) {
+    } else if ("file".equalsIgnoreCase(reference.getScheme())) {
```

Comment:
✏️ This is unnecessary, as the test on line 468 has already guaranteed that `reference.getScheme()` is non null. Sonar should absolutely _not_ have tagged this, regardless of the configuration applied to it.